src/ServiceControl.Plugin.Nsb5.Heartbeat/Heartbeats.cs

```diff
@@ -12,96 +12,159 @@
+        protected override void Setup(FeatureConfigurationContext context)
         {
-            if (!IsEnabledByDefault)
+        }
+
+        class HeartbeatStartup : FeatureStartupTask, IDisposable
+        {
+            public HeartbeatStartup(ISendMessages sendMessages, Configure configure, UnicastBus unicastBus)
             {
-                return;
+                this.unicastBus = unicastBus;
+
+                backend = new ServiceControlBackend(sendMessages, configure);
+                endpointName = configure.Settings.EndpointName();
+
+                var interval = ConfigurationManager.AppSettings[@"Heartbeat/Interval"];
+                if (!String.IsNullOrEmpty(interval))
+                {
+                    heartbeatInterval = TimeSpan.Parse(interval);
+                }
+
+                ttlTimeSpan = TimeSpan.FromTicks(heartbeatInterval.Ticks * 4); // Default ttl
+                var ttl = ConfigurationManager.AppSettings[@"Heartbeat/TTL"];
+                if (!String.IsNullOrWhiteSpace(ttl))
+                {
+                    if (TimeSpan.TryParse(ttl, out ttlTimeSpan))
+                    {
+                        Logger.InfoFormat("Heartbeat/TTL set to {0}", ttlTimeSpan);
+                    }
+                    else
+                    {
+                        ttlTimeSpan = TimeSpan.FromTicks(heartbeatInterval.Ticks * 4);
+                        Logger.Warn("Invalid Heartbeat/TTL specified in AppSettings. Reverted to default TTL (4 x Heartbeat/Interval)");
+                    }
+                }
             }
-            
-            backend = new ServiceControlBackend(SendMessages, Configure, CriticalError);
-            backend.VerifyIfServiceControlQueueExists();
-            heartbeatInterval = TimeSpan.FromSeconds(10); // Default interval
-            var interval = ConfigurationManager.AppSettings[@"Heartbeat/Interval"];
-            
-            if (!String.IsNullOrEmpty(interval))
+
+            protected override void OnStart()
             {
-                heartbeatInterval = TimeSpan.Parse(interval);
+                cancellationTokenSource = new CancellationTokenSource();
+
+                NotifyEndpointStartup(unicastBus.HostInformation, DateTime.UtcNow);
+                StartHeartbeats(unicastBus.HostInformation);
             }
 
-            ttlTimeSpan = TimeSpan.FromTicks(heartbeatInterval.Ticks * 4); // Default ttl
-            var ttl = ConfigurationManager.AppSettings[@"Heartbeat/TTL"];
-            if (!String.IsNullOrWhiteSpace(ttl))
+            protected override void OnStop()
             {
-                if (TimeSpan.TryParse(ttl, out ttlTimeSpan))
+                if (heartbeatTimer != null)
                 {
-                    logger.InfoFormat("Heartbeat/TTL set to {0}", ttlTimeSpan);
+                    heartbeatTimer.Dispose();
                 }
-                else
+
+                if (cancellationTokenSource != null)
```

Comment:
If `OnStart` isn't called / doesn't complete. Do you think we should ignore that case?